lib/backend/k8s/conversion/conversion_test.go

```diff
@@ -1639,13 +1639,96 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
 		Expect(pol.Value.(*apiv3.NetworkPolicy).Spec.Egress[1].Destination.NotNets[2]).To(Equal("192.168.5.0/24"))
 
 		// There should be no Ingress
-		Expect(len(pol.Value.(*apiv3.NetworkPolicy).Spec.Ingress)).To(Equal(0))
+		Expect(pol.Value.(*apiv3.NetworkPolicy).Spec.Ingress).To(HaveLen(0))
 
 		// Check that Types field exists and has only 'egress'
 		Expect(len(pol.Value.(*apiv3.NetworkPolicy).Spec.Types)).To(Equal(1))
 		Expect(pol.Value.(*apiv3.NetworkPolicy).Spec.Types[0]).To(Equal(apiv3.PolicyTypeEgress))
 	})
 
+	It("should parse a NetworkPolicyPeer with an IPBlock Peer and no Except field", func() {
+		np := networkingv1.NetworkPolicyPeer{
+			IPBlock: &networkingv1.IPBlock{
+				CIDR: "192.168.0.0/16",
+			},
+		}
+
+		// Parse the policy.
+		podSel, nsSel, nets, notNets := c.k8sPeerToCalicoFields(&np, "default")
+
+		// Assert value fields are correct.
+		Expect(nets[0]).To(Equal("192.168.0.0/16"))
+		Expect(notNets).To(BeNil())
+
+		// As this is a IPBlock rule, podSel and nsSel should be empty (not nil!)
```

Comment:
strings can't be nil, their "zero value" is the empty string.

Reply:
Yes, but in this case I'm checking against the array and not directly into the string :)